File: frontend/src/types/template.ts
export interface CertificateTemplate {
  id: number;
  name: string;
  description: string | null;
  original_filename: string;
  file_type: string;
  preview_image_path: string | null;
  page_width: number | null;
  page_height: number | null;
  created_at: string;
  updated_at: string;
}

export interface TemplateUploadResponse {
  template: CertificateTemplate;
  message: string;
}

File: backend/app/schemas/template.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class TemplateBase(BaseModel):
    name: str
    description: str | None = None


class TemplateRead(TemplateBase):
    id: int
    original_filename: str
    file_type: str
    preview_image_path: str | None = None
    page_width: float | None = None
    page_height: float | None = None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class TemplateUploadResponse(BaseModel):
    template: TemplateRead
    message: str
